### PR TITLE
angband, bump version

### PR DESCRIPTION
**Pull request by Begasus**
Builds ok on x86 (secondaryArch), not tested (yet) on x86_x64, fix for gcc2 would be minor (mostly parse errors) ...

### CODE REVIEW

File: games-roguelike/angband/patches/angband-4.1.2.patchset [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+diff --git a/src/z-rand.c b/src/z-rand.c
+index 57f0e4b..526af98 100644
+--- a/src/z-rand.c
++++ b/src/z-rand.c
+@@ -537,7 +537,9 @@ void rand_fix(u32b val)
+ 	rand_fixval = val;
+ }
+ 
++#ifndef __HAIKU__
+ int getpid(void);
```
Comment (alaviss) [inline games-roguelike/angband/patches/angband-4.1.2.patchset]: Actually, just changing `int` to `pid_t` should solve the problem for us and possibly others, as that's the [correct signature][0] for the function :)

```diff
-+#ifndef __HAIKU__
- int getpid(void);
+-int getpid(void);
++pid_t getpid(void);
-+#endif
```
[0]: http://pubs.opengroup.org/onlinepubs/9699919799/functions/getpid.html

File: games-roguelike/angband/angband-4.1.2.recipe [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	settings/angband/gamedata/slay.txt keep-old
+	settings/angband/gamedata/store.txt keep-old
+	settings/angband/gamedata/terrain.txt keep-old
+	settings/angband/gamedata/trap.txt keep-old
+	settings/angband/gamedata/vault.txt keep-old
+	settings/angband/gamedata/world.txt keep-old
+	"
+
+ARCHITECTURES="!x86_gcc2 x86 ?x86_64"
+SECONDARY_ARCHITECTURES="x86"
```
[the remark this answers is not in the document]
Comment (Begasus) [reply games-roguelike/angband/angband-4.1.2.recipe]: Started out to try to fix them, using commandBinDir would be the next step if no one disapproves :)

File: games-roguelike/angband/patches/angband-4.1.2.patchset [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+diff --git a/configure.ac b/configure.ac
+index 94ef3fe..66fdf2c 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -98,7 +98,7 @@ CFLAGS="$CFLAGS -DHAVE_CONFIG_H"
+ CPPFLAGS="$CPPFLAGS -I." 
+ 
+ if test "$GCC" = "yes"; then
+-	CFLAGS="$CFLAGS -W -Wall -Wextra -Wno-unused-parameter -pedantic"
++	CFLAGS="$CFLAGS -W -Wall"
```
Comment (alaviss) [inline games-roguelike/angband/patches/angband-4.1.2.patchset]: Why was the warning flags removed? Also, `-pedantic` is necessary for strict ISO C compilance, so I would prefer to keep it.

Comment (Begasus) [reply games-roguelike/angband/patches/angband-4.1.2.patchset]: I used some of the patch from previous version, probably due to not being able to build it on x86_gcc2 at that moment ... 
https://github.com/haikuports/haikuports/blob/75a3cf5384f650242f4b8a1094ab305d33ed99e7/games-roguelike/angband/patches/angband-3.5.0.patchset

Comment (Begasus) [reply games-roguelike/angband/patches/angband-4.1.2.patchset]: This isn't needed indeed, I think they were needed to get the previous build for x86_gcc2 working ....

**Pull request merged**